src/public_key.rs

```diff
@@ -381,6 +381,14 @@ impl PublicKey {
             PublicKeyRepr::Secp256k1(..) => secp256k1::PUBLIC_KEY_LENGTH,
         }
     }
+
+    pub fn to_solana(&self) -> Option<solana_sdk::pubkey::Pubkey> {
```

Comment:
Should this not be a TryFrom on PublicKey itself so it can return errors for the other types?